File: 00155-min-stack/solution.py
class MinStack:
    def __init__(self):
        self.stack = []
        self.min = []

    def push(self, val: int) -> None:
        if not self.min or self.min[-1] >= val:
            self.min.append(val)
        self.stack.append(val)

    def pop(self) -> None:
        if self.stack.pop() == self.min[-1]:
            self.min.pop()

    def top(self) -> int:
        return self.stack[-1]

    def getMin(self) -> int:
        return self.min[-1]

File: 00155-min-stack/solution.cs
public class MinStack {
    private Stack<int> stack = new Stack<int>();
    private Stack<int> min = new Stack<int>();

    public MinStack() {
    }

    public void Push(int val) {
        stack.Push(val);
        if (min.Count == 0 || val <= min.Peek()) {
            min.Push(val);
        }
    }

    public void Pop() {
        if (stack.Pop() == min.Peek()) {
            min.Pop();
        }
    }

    public int Top() {
        return stack.Peek();
    }

    public int GetMin() {
        return min.Peek();
    }
}
